WordPress/gutenberg
# Quick Edit: add Template field

## What?


This PR is based on top of https://github.com/WordPress/gutenberg/pull/66531
This PR is part of https://github.com/WordPress/gutenberg/issues/64519.


## Why?


This PR adds the `Template` field.


## Testing Instructions

1. Enable the DataViews specific experiments under Gutenberg > Experiments.
2. Make sure you have a block themed theme enabled.
3. Go to Appearance > Editor and click Pages.
4. Switch the page view to the table view.
5. Select a page and select the Details icon.
6. A Template field should now be displayed.
7. Click on the template field and ensure that a dropdown is visible.
8. Click on swap template.
9. Ensure that a modal is visible and all the templates are rendered correctly.
10. Click on a template.
11. Save.
12. Ensure that the page has as template the selected one.
13. Click on the template field and ensure that a dropdown is visible.
14. Click on `Use default template`.
15. Ensure that the default template is set.

### Testing Instructions for Keyboard


## Screenshots or screencast 

https://github.com/user-attachments/assets/5b278d3f-2f54-4465-a91a-d365a5d1b73c

## Drawbacks

### `@wordpress/block-editor` dependency

To implement the modal previewing all the templates is necessary use the `__experimentalBlockPatternsList` from the `@wordpress/block-editor` package. I think that it is acceptable.

### ⚠️ `Uncaught TypeError: Cannot destructure property 'avatarURL' of '(0 , _wordpress_data__WEBPACK_IMPORTED_MODULE_3__.useSelect)(...)' as it is undefined` error


In some cases, the templates in the modal aren't rendered correctly due to this JavaScript error:

```
Uncaught TypeError: Cannot destructure property 'avatarURL' of '(0 , _wordpress_data__WEBPACK_IMPORTED_MODULE_3__.useSelect)(...)' as it is undefined
```

https://github.com/user-attachments/assets/7950309e-31ce-4c2f-a173-cf9825fd4839

https://github.com/WordPress/gutenberg/blob/61f965533542941fd9ffe525933fa61c52464f00/packages/block-library/src/avatar/hooks.js#L19-L26

This happens when the setting object isn't initialized. To fix this issue, I pushed 5a623fbf0825298547b80f782e420610e6d61bab. Still, I'm not sure that this is the best solution 🤔.

It is a similar issue that I tried to tackle with #66459.

cc @oandregal @youknowriad and @louwie17
